packages/sourcecred/src/core/ledger/policies/balanced.js

```diff
@@ -54,41 +55,78 @@ export type BalancedPolicy = {|
  * scores.
  */
 export function balancedReceipts(
-  budget: G.Grain,
-  identities: ProcessedIdentities
+  policy: BalancedPolicy,
+  identities: ProcessedIdentities,
+  credGrainData: CredGrainView
 ): $ReadOnlyArray<GrainReceipt> {
-  const totalCred = sum(identities.map((x) => x.lifetimeCred));
-  const totalEverPaid = G.sum(identities.map((i) => i.paid));
+  if (policy.numIntervalsLookback < 0) {
+    throw new Error(
+      `numIntervalsLookback must be at least 0, got ${policy.numIntervalsLookback}`
+    );
+  }
+  if (!Number.isInteger(policy.numIntervalsLookback)) {
+    throw new Error(
+      `numIntervalsLookback must be an integer, got ${policy.numIntervalsLookback}`
+    );
+  }
+
+  const intervals = credGrainData.intervals();
+  if (
+    !policy.numIntervalsLookback ||
+    policy.numIntervalsLookback > intervals.length
+  ) {
+    policy.numIntervalsLookback = intervals.length;
```

Comment:
set to a new const to avoid overwriting params

Reply:
Fixed. Did this as an arrow function, not sure if there is a more elegant strategy.